cosmos/config.py

```diff
@@ -257,6 +259,36 @@ def validate_profiles_yml(self) -> None:
         if self.profiles_yml_filepath and not Path(self.profiles_yml_filepath).exists():
             raise CosmosValueError(f"The file {self.profiles_yml_filepath} does not exist.")
 
+    def _handle_profile_caching(self, desired_profile_path: Path | None = None, use_mock_values: bool = False) -> Path:
```

Comment:
Could we find a name that represents better what this method does? Ultimately, it returns the profile path.
And the docstring could also give a brief description on what it does, before detailing the how, something along the lines:
`Return previously cached profile path or calculate and store the profile content into a file and return its path`